neo4j/api.py

```diff
@@ -172,23 +174,44 @@ def values(self):
 
 
 class ServerInfo:
+    """ Represents a package of information relating to a Neo4j server.
+    """
 
     def __init__(self, address, protocol_version):
-        self.address = address
-        self.protocol_version = protocol_version
-        self.metadata = {}
+        self._address = address
+        self._protocol_version = protocol_version
+        self._metadata = {}
+
+    @property
+    def address(self):
+        """ Network address of the remote server.
+        """
+        return self._address
+
+    @property
+    def protocol_version(self):
+        """ Bolt protocol version with which the remote server
+        communicates.
+        """
+        return self._protocol_version
 
     @property
     def agent(self):
-        """The server agent string the server responded with.
+        """ Server agent string by which the remote server identifies
+        itself.
+        """
+        return self._metadata.get("server")
 
-        :return: Server agent string
-        :rtype: str
+    @property
+    def connection_id(self):
+        """ Unique identifier for the remote server connection.
         """
-        # Example "Neo4j/4.0.5"
-        # Example "Neo4j/4"
-        return self.metadata.get("server")
+        return self._metadata.get("connection_id")
 
+    # TODO in 5.0: remove this method
+    @deprecated("The version_info method is deprecated, please use "
+                "ServerInfo.agent, ServerInfo.protocol_version, or "
+                "call the dbms.components procedure instead")
```

Comment:
Isn't dbms.components always the appropriate action?

Reply:
It depends. Due to how the version information has been collected historically, there may be several reasons why users are calling this method. Drawing attention to all methods can only help.